src/hooks/useCountDown.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+
+const MS = 1000;
+
+export function useCountdown(
+	seconds: number,
+	setSeconds: (s: number) => void
+) {
+	useEffect(() => {
+		if (seconds <= 0) { return; }
+		const timer = setTimeout(() => {
+			setSeconds(seconds - 1);
+		}, MS);
+		return () => clearTimeout(timer);
+	}, [seconds, setSeconds]);
+
+	return { isFinished: seconds <= 0 };
+}
```

Comment:
```suggestion
import { useEffect, useState } from 'react';

const ONE_SECOND_IN_MS = 1000;

export function useCountdown(seconds: number) {
	const [remaningTime, setRemainingTime] = useState(seconds);

	useEffect(() => {
		const interval = setInterval(() => {
			setRemainingTime(remainingTime - 1);

			if (remainingTime <= 0) {
				clearInterval(interval);
			}
		}, seconds * ONE_SECOND_IN_MS);
	}, [seconds]);

	return { isFinished: remainingTime <= 0, remainingTime };
}
```

- Use `setInterval` instead of `setTimeout`, this is what the function is for.
- Keep the state _inside_ the hook.
- Return the remaining time, it is data from the hook and not external to it.